src/document.rs

```diff
@@ -33,14 +55,139 @@ impl<'a> DocumentEncryptOpts {
         DocumentEncryptOpts {
             id,
             name,
-            grant_to_author,
-            grants,
+            grants: EitherOrBoth::Left(ExplicitGrant {
+                grants,
+                grant_to_author,
+            }),
         }
     }
+
+    pub fn with_policy_grants(
+        id: Option<DocumentId>,
+        name: Option<DocumentName>,
+        policy: PolicyGrant,
+    ) -> DocumentEncryptOpts {
+        DocumentEncryptOpts {
+            id,
+            name,
+            grants: EitherOrBoth::Right(policy),
+        }
+    }
+}
+
+/// Document access granted by a policy.
+///
+/// A policy is stored on the server and yields a list of users and groups when applied.
+///
+/// `category` -
+/// `sensitivity` -
+/// `data_subject` -
+/// `substitute_id` -
+#[derive(Debug, PartialEq, Clone)]
+pub struct PolicyGrant {
+    category: Option<Category>,
+    sensitivity: Option<Sensitivity>,
+    data_subject: Option<DataSubject>,
+    substitute_user_id: Option<SubstituteId>,
+}
+
+impl PolicyGrant {
+    pub fn new(
+        category: Option<Category>,
+        sensitivity: Option<Sensitivity>,
+        data_subject: Option<DataSubject>,
+        substitute_user: Option<UserId>,
+    ) -> Result<PolicyGrant> {
+        if let (None, None, None, None) = (&category, &sensitivity, &data_subject, &substitute_user)
+        {
+            Err(IronOxideErr::InvalidPolicy)
+        } else {
+            Ok(PolicyGrant {
+                category,
+                sensitivity,
+                data_subject,
+                substitute_user_id: substitute_user.map(|u| u.into()),
+            })
+        }
+    }
+
+    pub fn category(&self) -> Option<&Category> {
+        self.category.as_ref()
+    }
+
+    pub fn sensitivity(&self) -> Option<&Sensitivity> {
+        self.sensitivity.as_ref()
+    }
+
+    pub fn data_subject(&self) -> Option<&DataSubject> {
+        self.data_subject.as_ref()
+    }
+    pub fn substitute_id(&self) -> Option<&SubstituteId> {
+        self.substitute_user_id.as_ref()
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct Category(pub(crate) String);
+
+impl TryFrom<&str> for Category {
+    type Error = IronOxideErr;
+
+    fn try_from(value: &str) -> Result<Self> {
+        validate_simple_policy_id(value, "Category").map(|v| Self(v))
+    }
+}
+
+impl Category {
+    pub(crate) const QUERY_PARAM: &'static str = "category";
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct Sensitivity(pub(crate) String);
+
+//TODO macro?
```

Comment:
Meh, I'd just remove the comment and move on.

Reply:
Macro added as it was easy.